_studio/shared/umc/codec/vc1_dec/include/umc_vc1_video_decoder_hw.h

```diff
@@ -157,10 +157,19 @@ namespace UMC
 
             m_bLastFrameNeedDisplay = true;
 
-            if (VC1_SKIPPED_FRAME == pPackDescriptorChild->m_pContext->m_picLayerHeader->PTYPE)
-                out_data->SetFrameType(D_PICTURE); // means skipped 
-            else
-                out_data->SetFrameType(I_PICTURE);
+            switch (pPackDescriptorChild->m_pContext->m_picLayerHeader->PTYPE & VC1_BI_FRAME)
+            {
+                case  VC1_I_FRAME:
+                    out_data->SetFrameType(I_PICTURE);
+                    break;
+                case  VC1_P_FRAME:
+                    out_data->SetFrameType(P_PICTURE);
+                    break;
+                case  VC1_B_FRAME:
+                case  VC1_BI_FRAME:
+                    out_data->SetFrameType(B_PICTURE);
+                    break;
```

Comment:
You know, I am nervous when I see switch without the default case. This is origin of errors. Hence this is the wonderful place to add `assert()` even if you believe we won't be here.